docs/external-db.md

```diff
@@ -0,0 +1,159 @@
+# External DB integration
+
+Backstage hosts the data in the [PostgreSQL database](https://backstage.io/docs/getting-started/config/database/).
+By default, Operator creates and manages local instance of PostgreSQL in the same as Backstage deployment namespace but it also allows to switch this off and configure external instance instead.
+Usually, external connection requires more security, so, this instruction includes steps to configure SSL/TLS.
+
+### Configure your external PostgreSQL instance
+As a prerequisite you have to know:
+- **db-host** - your PostgreSQL instance DNS or IP address 
+- **db-port** - your PostgreSQL instance port number (usually 5432)
+- **username** - to connect to your PostgreSQL instance
+- **password** - to connect to your PostgreSQL instance
+
+In addition, to get your database connection secured with SSL/TLS you also need certificates in a form of PEM file. 
+
+You can find configuration guidelines for:
+- [AWS RDS PostgreSQL](#aws-rds-postgresql)
+- [Azure Database PostgreSQL](#azure-db-postgresql)
+
+### Create secret with PostgreSQL connection properties:
+````
+cat <<EOF | kubectl -n <your-namespace> create -f -
+apiVersion: v1
+kind: Secret
+metadata:
+ name: <cred-secret-name>
+type: Opaque
+stringData:
+ POSTGRES_PASSWORD: <password>
+ POSTGRES_PORT: <db-port>
+ POSTGRES_USER: <username>
+ POSTGRES_HOST: <db-host>
+ PGSSLMODE: require #  for TLS connection
+ NODE_EXTRA_CA_CERTS: <abs-path-to-pem-file> # for TLS connection, e.g. /opt/app-root/src/postgres-crt.pem
+EOF
+````
+
+### Create secret with certificate(s):
+(omit this step if you do not need TLS connection, maybe for testing purpose)
+
+````
+cat <<EOF | kubectl -n <your-namespace> create -f -
+apiVersion: v1
+kind: Secret
+metadata:
+ name: <crt-secret>
+type: Opaque
+stringData:
+ postgres-crt.pem: |-
+   -----BEGIN CERTIFICATE-----
+   MIIFqDCCA5CgAwIBAgIQHtOXCV/YtLNHcB6qvn9FszANBgkqhkiG9w0BAQwFADBl
+   ... 
+````
+
+### Create Backstage Custom Resource:
+
+- disable creating local PostgreSQL instance with **spec.database.enableLocalDb: false**
+- add **<crt-secret>** to **spec.application.extraFiles.secrets**, so, as for example below **postgres-crt.pem** file will be mounted to Backstage container at **spec.application.extraFiles.mountPath** directory:   
+- add **<cred-secret>** to **spec.application.extraEnvs.secrets**, so all the data's entries will be injected to Backstage container as environment variables.
```

Comment:
I know that it would achieve the same result, but how about using the [`spec.database.authSecretName` field](https://github.com/janus-idp/operator/blob/main/api/v1alpha1/backstage_types.go#L68) instead? As the name suggests, it is the secret for database authentication, so I would expect this info to be defined there.

Reply:
Yes, it would achieve the same result and talking to users I realised they tend to use the extra-secrets way as a generic approach to inject env variables (not necessarily for app-config), that's why I decided to promote this way.